src/proto/Cardano.proto

```diff
@@ -42,20 +50,47 @@ message Transfer {
     // Change address
     string change_address = 2;
 
-    // ADA amount to transfer
+    // Requested ADA amount to be transferred. Output can be different only in use_max_amount case.
+    // Note that Cardano has a minimum amount per UTXO, see TWCardanoMinAdaAmount.
     uint64 amount = 3;
 
-    // Set if max amount is requested (max possible from UTXOs/fee; amount is disregarded in this case)
-    bool use_max_amount = 4;
+    // Optional token(s) to be transferred
+    // Currently only one token type to be transferred per transaction is supported
+    TokenBundle token_amount = 4;
+
+    // Request max amount option. If set, tx will send all possible amounts from all inputs, including all tokens; there will be no change; requested amount and token_amount is disregarded in this case.
+    bool use_max_amount = 5;
+
+    // Optional fee overriding. If left to 0, optimal fee will be calculated. If set, exactly this value will be used as fee.
+    // Use it with care, it may result in underfunded or wasteful fee.
+    uint64 force_fee = 6;
```

Comment:
Does this conflict with `fee` in `TransactionPlan`?

Reply:
No, it's usage is similar to `byteFee` in Bitcoin (though there that's not absolute fee amount):
- If plan is present, it is used, it determines the exact output amounts (incl. change), and thus implicitly the fee
- If plan is not amount, plan() will be executed, and if input.force_fee is set, that amount will be used.  If not, fee will be computed.

The UI can have a fee setting, populated with the computed fee value, and user can set it.  This could be useful in some tricky cases when the default fee is not valid for some reason.